### ISSUE TITLE
Alternative Texture Filtering

### ISSUE THREAD

**Issue by kettek** [AllenDang/giu#75]
It'd be nice to support texture filtering other than linear. This is easy enough to implement within `RendererOpenGL3` by adding the following to the drawing of Commands:

```
gl.TexParameteri(gl.TEXTURE_2D, gl.TEXTURE_MIN_FILTER, gl.NEAREST) // minification filter
gl.TexParameteri(gl.TEXTURE_2D, gl.TEXTURE_MAG_FILTER, gl.NEAREST) // magnification filter
```

However, this makes it global which isn't quite as nice as having filtering per-texture or per-Image. Digging into the code, it is non-trivial to add per image calls specify filtering, as a large portion of imgui's code has to be adjusted for extra min and mag fields.

So, as far as I can tell there are two options:
  * Add MinFilter and MagFilter fields/parameters to `imgui.NewOpenGL3(...)` and have the filtering work globally for all textures.
  * Add MinFilter and MagFilter fields/accessors/etc. to `imgui.DrawCommand` and `ImDrawCmd` to have filtering work per DrawCommand.

The first option is by far the easiest, as changes only need to be made to `RendererOpenGL3.go` and `MasterWindow.go`. The second option is fairly daunting but doable, though it'd make it harder to integrate future versions of imgui.

Regardless of whichever is best, it would be useful to have the first option implemented (especially for my uses which is pixel art). I'll make a pull request for this at the very least.

**Comment by AllenDang** [comment]
@kettek Thanks very much! I'm not quite familiar with opengl frankly. It's very nice to have someone to enforce that part. Looking forward to your pr.

**Comment by kettek** [comment]
See #76

**Comment by kettek** [comment]
Closed with merge of #76
